reference.py

```diff
@@ -98,20 +98,24 @@ def point_negate(P: Optional[Point]) -> Optional[Point]:
         return P
     return (x(P), p - y(P))
 
-def cpoint(x: bytes) -> Point:
+# parses compressed point (33-bytes array) into the `Point` type
+# Returns `None` for invalid inputs
+def cpoint(x: bytes) -> Optional[Point]:
     if len(x) != 33:
         raise ValueError('x is not a valid compressed point.')
     P = lift_x(int_from_bytes(x[1:33]))
     if P is None:
-        raise ValueError('x is not a valid compressed point.')
+        # invalid x-coordinate
+        return None
     if x[0] == 2:
         return P
     elif x[0] == 3:
         P = point_negate(P)
         assert P is not None
         return P
     else:
-        raise ValueError('x is not a valid compressed point.')
+        # invalid parity
+        return None
```

Comment:
Is it possible to avoid replacing the exception with None? It would be good to stay consistent with the BIPs. This change would introduce a difference between cpoint as defined in bip 327 and here.

Alternatively, we could somehow use https://github.com/BlockstreamResearch/bip-frost-dkg/tree/master/python/secp256k1proto and define a `cpoint` function there, such that all future BIPs share the same primitives. By the way, the library has a proper representation of infinity instead of representing infinity as None.

Reply:
Ahh yes. This is a terrible fix. 
I did this because `vector5()`, `vector6()`, and `vector7()` makes `cpoint` throw a value error. Unlike BIP327, our testing infra doesn't support vectors that throw errors.

Reply:
I'll implement the [`assert_raises`](https://github.com/bitcoin/bips/blob/master/bip-0327/reference.py#L475) utility fn to handle these.

Reply:
Fixed in a777151

Reply:
Can you make `schnorr_presig_verify` not return an exception but instead return False?

Reply:
Done! Now `extract_adaptor` returns `None` if `cpoint` raises value error.